### PR TITLE
Clean up source to get Linux builds working

### PR DESCRIPTION
**Pull request by andrewc12**
### Motivation and Context

In my quest to improve the testing situation for windows I accidently stumbled across the idea of cleaning up the source to be able to get both windows and linux builds working.
This should let us use both the automated linux builds to test for large breaking changes and serve as the groundwork for either using the linux tests under wsl or to port the tests to natively to windows.

### Description

- moving windows specific code into windows specific source files
- enclosing windows specific code in conditionals
- creating defines for new functions in other os source files
- changing some types to macros that support multiple os 

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [ ] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [x] I have run the ZFS Test Suite with this change applied.
- [ ] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by andrewc12** [comment]
@datacore-rm you might want to give this a look to see if it causes problems with perfmon counters.

**Comment by datacore-rm** [comment]
Changes looks fine.Thanks.

File: module/zfs/zvol.c [hunk 1]
```diff
@@ -1740,6 +1740,9 @@ zvol_fini_impl(void)
 }
 
 /* ZFS ZVOLDI */
+// HACK FIXME
```
Comment (lundman) [inline module/zfs/zvol.c:1743]: Shouldn't these be in `zvol_os.c` ?

File: module/zfs/dsl_pool.c [hunk 2]
```diff
@@ -117,7 +117,7 @@ ZFS_MODULE_ULONG zfs_wrlog_data_max = 0;
  * zfs_dirty_data_max), push out a txg.  This should be less than
  * zfs_vdev_async_write_active_min_dirty_percent.
  */
```
Comment (lundman) [inline module/zfs/dsl_pool.c:119]: we are removing static?

File: include/sys/vdev_file.h [hunk 3]
```diff
@@ -35,6 +35,11 @@ typedef struct vdev_file {
 	uint64_t	vdev_win_offset; /* soft partition start */
 	uint64_t	vdev_win_length; /* soft partition length */
 } vdev_file_t;
+#else
+/* Grabbed from https://github.com/openzfs/zfs/blob/1d3ba0bf01020f5459b1c28db3979129088924c0/include/sys/vdev_file.h#L36 */
+typedef struct vdev_file {
+    zfs_file_t* vf_file;
```
Comment (lundman) [inline include/sys/vdev_file.h:41]: nit: "<space>*vf_file;"

File: cmd/zpool/os/linux/zpool_vdev_os.c [hunk 4]
```diff
@@ -417,9 +417,3 @@ check_file(const char *file, boolean_t force, boolean_t isspare)
 {
 	return (check_file_generic(file, force, isspare));
 }
-
```
Comment (lundman) [inline cmd/zpool/os/linux/zpool_vdev_os.c:420]: there were two? hah neat

**Pull request merged**